aten/src/ATen/native/Cross.cpp

```diff
@@ -42,9 +43,41 @@ Tensor & cross_out(const Tensor & input, const Tensor & other, const c10::option
     TORCH_CHECK(input.size(dim) == 3, "dimension ", dimension.value(), " does not have size 3");
   }
 
-  if (out.sizes() != input.sizes()) {
-    out.resize_as_(input);
-  }
+  // check if resizing output is required
+  // raise a warning while resizing if output has one or more elements
+  at::native::resize_output(out, input.sizes());
+
+  cross_stub(device1, out, input, other, dim);
+  return out;
+}
+
+Tensor linalg_cross(const Tensor & input, const Tensor & other, const int64_t dimension) {
+  Tensor out = at::empty_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
Not that it changes much, but in this case, but we often do:
```suggestion
  Tensor out = at::empty({0}, input.options());
```
and delegate to the `_out` version the creation of the tensor. This plays better with operations in which the creation of the `out` tensor is non-trivial.

Reply:
Done.